sdk/eventhub/event-hubs/src/partitionPump.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as log from "./log";
+import { EventProcessorOptions, PartitionProcessor } from "./eventProcessor";
+import { PartitionContext } from "./partitionContext";
+import { EventHubClient } from "./eventHubClient";
+import { EventPosition } from "./eventPosition";
+import { EventHubConsumer } from "./receiver";
+
+export class PartitionPump {
+  private _partitionContext: PartitionContext;
+  private _eventHubClient: EventHubClient;
+  private _partitionProcessor: PartitionProcessor;
+  private _processorOptions: EventProcessorOptions;
+  private _receiver: EventHubConsumer | undefined;
+  private _isReceiving: boolean = false;
+
+  constructor(
+    eventHubClient: EventHubClient,
+    partitionContext: PartitionContext,
+    partitionProcessor: PartitionProcessor,
+    options?: EventProcessorOptions
+  ) {
+    if (!options) options = {};
+    this._eventHubClient = eventHubClient;
+    this._partitionContext = partitionContext;
+    this._partitionProcessor = partitionProcessor;
+    this._processorOptions = options;
+  }
+
+  async start(partitionId: string): Promise<void> {
+    await this._partitionProcessor.initialize!();
```

Comment:
Since we're stating that `initialize` is optional, we should do a type check first to see if it is a function. If it is, then we can call it.

Reply:
Should we also add the same check for required parameters i.e. `processEvents` and `processError` before calling it?

Reply:
I think we could probably do validation when we create the PartitionProcessor to ensure that the required methods are there. If they aren't we could throw an error then. If we do that, we don't need to do the check here.